auth_password_settings/models/res_users.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Dhaval Patel
+#    Copyright (C) 2011 - TODAY Denero Team. (<http://www.deneroteam.com>)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.osv import osv
+from openerp.tools.safe_eval import safe_eval
+from openerp.tools.translate import _
+import string
+
+
+class res_users(osv.osv):
+    _inherit = "res.users"
+
+    def _validate_password(self, cr, uid, password, context=None):
+        icp = self.pool.get('ir.config_parameter')
+        password_rules = []
+        config_data = {
+            'auth_password_min_character': safe_eval(
+                icp.get_param(
+                    cr,
+                    uid,
+                    'auth_password_settings.auth_password_min_character',
+                    '6'
+                )),
+            'auth_password_has_capital_letter': safe_eval(
+                icp.get_param(
+                    cr,
+                    uid,
+                    'auth_password_settings.auth_password_has_capital_letter',
+                    'False'
+                )),
+            'auth_password_has_digit': safe_eval(
+                icp.get_param(
+                    cr,
+                    uid,
+                    'auth_password_settings.auth_password_has_digit',
+                    'False'
+                )),
+            'auth_password_has_special_letter': safe_eval(
+                icp.get_param(
+                    cr,
+                    uid,
+                    'auth_password_settings.auth_password_has_special_letter',
+                    'False'
+                )),
+        }
+        password_rules.append(
+            lambda s:
+                len(s) >= config_data.get('auth_password_min_character', 6) or
```

Comment:
don't use `dict.get(key)` when you are sure the key is present (and here you are). Use `dict[key]`